tests/test_export.py

```diff
@@ -383,3 +383,30 @@ def test_2121_wrapped_self_serialize(snippetcompiler):
     )
     with pytest.raises(TypeError, match="not JSON serializable"):
         snippetcompiler.do_export()
+
+
+def test_3787_key_error_export(snippetcompiler):
+    """
+    Check the error message of an export with a KeyError
+    The Key error happens in get_real_name() of class Test3
+    """
+    snippetcompiler.setup_for_snippet(
```

Comment:
So you're right that this is generally the way to test this. If you think it would be better to have this a bit more visible you could use (but perhaps only for iso5 + master) `tests/util.py::module_from_template` instead. It would require `snippetcompiler_clean` though because it installs the module in the snippetcompiler's venv.